contributing-docs/11_provider_packages.rst

```diff
@@ -0,0 +1,235 @@
+=================
+
+Airflow 2.0 is split into core and providers. They are delivered as separate packages:
+
+* ``apache-airflow`` - core of Apache Airflow
+* ``apache-airflow-providers-*`` - More than 70 provider packages to communicate with external services
+
+.. contents:: :local:
+
+Where providers are kept in our repository
+------------------------------------------
+
+Airflow Providers are stored in the same source tree as Airflow Core (under ``airflow.providers``) package. This
+means that Airflow's repository is a monorepo, that keeps multiple packages in a single repository. This has a number
+of advantages, because code and CI infrastructure and tests can be shared. Also contributions are happening to a
+single repository - so no matter if you contribute to Airflow or Providers, you are contributing to the same
+repository and project.
+
+It has also some disadvantages as this introduces some coupling between those - so contributing to providers might
+interfere with contributing to Airflow. Python ecosystem does not yet have proper monorepo support for keeping
+several packages in one repository and being able to work on multiple of them at the same time, but we have
+high hopes Hatch project that use as our recommended packaging frontend
+will `solve this problem in the future <https://github.com/pypa/hatch/issues/233>`__
+
+Therefore, until we can introduce multiple ``pyproject.toml`` for providers information/meta-data about the providers
+is kept in ``provider.yaml`` file in the right sub-directory of ``airflow\providers``. This file contains:
+
+* package name (``apache-airflow-provider-*``)
+* user-facing name of the provider package
+* description of the package that is available in the documentation
+* list of versions of package that have been released so far
+* list of dependencies of the provider package
+* list of additional-extras that the provider package provides (together with dependencies of those extras)
+* list of integrations, operators, hooks, sensors, transfers provided by the provider (useful for documentation generation)
+* list of connection types, extra-links, secret backends, auth backends, and logging handlers (useful to both
+  register them as they are needed by Airflow and to include them in documentation automatically).
+* and more ...
+
+If you want to add dependencies to the provider, you should add them to the corresponding ``provider.yaml``
+and Airflow pre-commits and package generation commands will use them when preparing package information.
+
+In Airflow 1.10 all those providers were installed together within one single package and when you installed
+airflow locally, from sources, they were also installed. In Airflow 2.0, providers are separated out,
+and not packaged together with the core when you build "apache-airflow" package, however when you install
+airflow project in editable mode with ``pip install -e ".[devel]"`` they are available in the same
+environment as Airflow.
```

Comment:
Airflow 1.10 is sooo legacy, do you think somebody still cares about this history?
```suggestion
In Airflow 2.0, providers are separated out, and not packaged together with the core when
you build "apache-airflow" package, however when you install airflow project in editable
mode with ``pip install -e ".[devel]"`` they are available in the same environment as Airflow.
```